### ISSUE TITLE
Make YAML files extensions configurable

### ISSUE THREAD

**Issue by gdubicki** [adrienverge/yamllint#85]
In some solutions YAML files use different extensions than `.yml` or `.yaml`.

For example Puppet software configuration management tool can use YAML files as a source of configuration. When using solution called hiera-eyaml (https://github.com/voxpupuli/hiera-eyaml) to keep some of the configuration values (such as passwords) encrypted those YAML files have `.eyaml` extension.

It would be then nice to make the extensions used by `yamllint .` configurable, while keeping the default as `.yml` and `.yaml`.

**Comment by adrienverge** [comment]
Hey,

This is a good idea.

I see another similar use-case: lint files according to their path, not their
extension. For instance with Ansible, some config files don't have an extension
(e.g. `env/prod/group_vars/all/config`) but are YAML files anyway.

Do you think adding something like this in the conf would do the job?

```yaml
yaml-files: |
  *.yaml
  *.yml

ignore: |
  *.dont-lint-me.yaml

extends: default
```

With hiera-eyaml, you would use:
```yaml
yaml-files: |
  *.e?ya?ml
```

And with Ansible:
```yaml
yaml-files: |
  *.ya?ml
  /env/*/group_vars/all/config
```

(Previous examples use a [`.gitignore`-style pattern matching](https://pypi.python.org/pypi/pathspec).)

**Comment by gdubicki** [comment]
Yes, that would be good!

I won’t create PR here though as I can live with using find and then yamllint here and I want to finish that “empty-value” rule.

**Comment by adrienverge** [comment]
Sure, I understand. Let's leave this issue open in case someone wants to implement it.

**Comment by craSH** [comment]
Hi there, just thought I'd leave a note as I would also like this feature! I specifically am starting to use this for some custom tooling as well as with Salt Stack (very similar use case as to Ansible).

For anyone with some extra free time, here's where to start digging for putting together a PR! :)
https://github.com/adrienverge/yamllint/blob/master/yamllint/cli.py#L35

**Comment by grzesuav** [comment]
Added in similar way as `ignore` was handled, please review : https://github.com/adrienverge/yamllint/pull/193

**Comment by adrienverge** [comment]
This is now implemented.
